app/controllers/candidate_interface/gcse/english/grade_controller.rb

```diff
@@ -6,31 +6,87 @@ class Gcse::English::GradeController < Gcse::DetailsController
     before_action :set_subject
 
     def edit
-      @application_qualification = details_form
-      @qualification_type = details_form.qualification.qualification_type
+      @gcse_grade_form = english_gcse_grade_form
+      @qualification_type = gcse_english_qualification.qualification_type
+
+      render view_path
     end
 
     def update
-      @qualification_type = details_form.qualification.qualification_type
-
-      details_form.grade = details_params[:grade]
-      details_form.other_grade = details_params[:other_grade]
+      @qualification_type = gcse_english_qualification.qualification_type
+      if multiple_gsces_are_active? && @qualification_type == 'gcse'
+        @gcse_grade_form = english_gcse_grade_form.assign_values(english_details_params)
+      else
```

Comment:
Would it be possible to move the attribute assignment in lines 20 - 23 to the form as you have done on line 18?

Reply:
Great idea, I think this helps to tidy up the controller. I've pushed the change now